src/indexer/index.ts

```diff
@@ -262,3 +268,43 @@ export async function getAssetInformation(
 		throw e;
 	}
 }
+
+/**
+ * Get parsed DAO information
+ * @param params object to decode
+ */
+export async function decodePSQLAppParams(params: any): Promise<DaoTableData> {
+	const appParams = JSON.parse(params.appParams);
+	const globalState = decodeAppParamsState(appParams.dt.gd);
+	const tokenData = await getAssetInformation(params.assetId);
+	return {
+		dao_id: +params.appId,
+		token_id: +params.assetId,
+		token_name: tokenData.an as string,
+		name: globalState.get(GLOBAL_STATE_MAP_KEY.DaoName) as string,
+		link: globalState.get(GLOBAL_STATE_MAP_KEY.Url) as string,
+	};
+}
+
+/**
+ * Get dao information
+ * @param searchDaoType parameter to query
+ * @param value paramter value
+ */
+export async function handleDaoSearch(
```

Comment:
I would suggest to keep this method in AllDao.vue file since it will be used there only. Other component won't be using it.

Reply:
actually it will be used in AllDao as well as AppHeader, currently I have used it in AppHeader only